lens/plotting.py

```diff
@@ -60,6 +63,161 @@ def plot_distribution(ls, column, bins=None):
+    normalised to the total number of non-null observations. For two
+    categorical variables, the plot produced is a heatmap representation of
+    the contingency table.
+
+    Parameters
+    ----------
+    ls : :class:`~lens.Summary`
+        Lens `Summary`.
+    column1 : str
+        First column.
+    column2 : str
+        Second column.
+
+    Returns
+    -------
+    :class:`plt.Figure`
+        Matplotlib figure containing the pairwise density plot.
+    """
+    pair_details = ls.pair_details(column1, column2)
+    pairdensity = pair_details['pairdensity']
+
+    x = np.array(pairdensity['x'])
+    y = np.array(pairdensity['y'])
+    Z = np.array(pairdensity['density'])
+
+    fig, ax = plt.subplots()
+
+    if ls.summary(column1)['desc'] == 'categorical':
+        idx = np.argsort(x)
+        x = x[idx]
+        Z = Z[:, idx]
+        # Create labels and positions for categorical axis
+        x_labels = dict(enumerate(x))
+        _set_integer_tick_labels(ax.xaxis, x_labels)
+        x = np.arange(-0.5, len(x), 1.0)
+
+    if ls.summary(column2)['desc'] == 'categorical':
+        idx = np.argsort(y)
+        y = y[idx]
+        Z = Z[idx]
+        y_labels = dict(enumerate(y))
+        _set_integer_tick_labels(ax.yaxis, y_labels)
+        y = np.arange(-0.5, len(y), 1.0)
+
+    X, Y = np.meshgrid(x, y)
+
+    ax.pcolormesh(X, Y, Z, cmap=DEFAULT_COLORSCALE.lower())
+
+    ax.set_xlabel(column1)
+    ax.set_ylabel(column2)
+
+    ax.set_title(r'$\it{{ {} }}$ vs $\it{{ {} }}$'.format(column1, column2))
+
+    return fig
+
+
+def plot_correlation_mpl(ls, include=None, exclude=None):
+    """Plot the correlation matrix for numeric columns
+
+    Plot a Spearman rank order correlation coefficient matrix showing the
+    correlation between columns. The matrix is reordered to group together
+    columns that have a higher correlation coefficient.  The columns to be
+    plotted in the correlation plot can be selected through either the
+    ``include`` or ``exclude`` keyword arguments. Only one of them can be
+    given.
+
+    Parameters
+    ----------
+    ls : :class:`~lens.Summary`
+        Lens `Summary`.
+    include : list of str
+        List of columns to include in the correlation plot.
+    exclude : list of str
+        List of columns to exclude from the correlation plot.
+
+    Returns
+    -------
+    :class:`plt.Figure`
+        Matplotlib figure containing the pairwise density plot.
+    """
+
+    columns, correlation_matrix = ls.correlation_matrix(include, exclude)
+    num_cols = len(columns)
+
+    if num_cols > 10:
+        annotate = False
+    else:
+        annotate = True
+
+    fig, ax = plt.subplots()
+    sns.heatmap(correlation_matrix, annot=annotate, fmt='.2f', ax=ax,
+                xticklabels=columns, yticklabels=columns, vmin=-1, vmax=1,
+                cmap='RdBu_r', square=True)
+
+    ax.xaxis.tick_top()
+
+    w = len(columns) * 2.5
+    while w > 10:
+        w /= np.sqrt(1.4)
```

Comment:
Some pretty heavy use of magic numbers here. Why those numbers? What does this do?